op-chain-ops/ether/migrate.go

```diff
@@ -194,5 +222,21 @@ func MigrateLegacyETH(db ethdb.Database, addresses []common.Address, allowances
 	// Set the total supply to 0
 	stateDB.SetState(predeploys.LegacyERC20ETHAddr, getOVMETHTotalSupplySlot(), common.Hash{})
 
-	return nil
+	if !commit {
+		log.Info("dry run, skipping commit")
+		return common.Hash{}, nil
+	}
+
+	log.Info("committing state DB")
+	newRoot, err := stateDB.Commit(true)
+	if err != nil {
+		return common.Hash{}, err
+	}
+
+	log.Info("committing trie DB")
+	if err := stateDB.Database().TrieDB().Commit(newRoot, true, nil); err != nil {
```

Comment:
This was missing previously, which meant that no state data was being committed to disk.